.github/workflows/kubeflow-release.yaml

```diff
@@ -0,0 +1,140 @@
+---
+name: Kubeflow Release Pipeline
+on:
+  workflow_dispatch:
+    inputs:
+      create-new-release:
+        description: "Create a new release?"
+        required: true
+        default: "true"
+env:
+  CREATE_NEW_RELEASE: ${{ inputs.create-new-release }}
+  REPO_OWNER: opendatahub-io
+  REPO_NAME: kubeflow
+  BRANCH_NAME: v1.9-branch
+
+jobs:
+  # 1. Sync changes to opendatahub:v1.9-branch from opendatahub:main
+  sync-main-to-release-branch:
+    uses: opendatahub-io/kubeflow/.github/workflows/sync-branches.yaml@main
+    with:
+      source: "main"
+      target: "v1.9-branch"
+
+  # 2. Poll for images to be available on quay.io the readiness of the images usually takes ~10 mins
+  wait-images-to-build-on-quay:
+    needs: sync-main-to-release-branch
+    runs-on: ubuntu-latest
+    outputs:
+      images_ready: ${{ steps.check-images.outputs.images_ready }}
+    steps:
+      - name: Poll for images availability
+        id: check-images
+        run: |
```

Comment:
Again, personal preference here (so fine to ignore) - but if we are going to inline scripts that aren't a couple lines... pulling that out into a proper external script seems easier to work with (i.e. don't need to worry about the lurking concern of YAML indentation, etc)

However, doing so requires the `actions/checkout@v4` workflow.. and I'm not clear on the potential performance concerns (if any) of introducing that.